src/firejail/preproc.c

```diff
@@ -58,6 +58,10 @@ void preproc_build_firejail_dir(void) {
 		create_empty_dir_as_root(RUN_FIREJAIL_X11_DIR, 0755);
 	}
 
+	if (stat(RUN_FIREJAIL_DBUS_DIR, &s)) {
+		create_empty_dir_as_root(RUN_FIREJAIL_DBUS_DIR, 01777);
```

Comment:
I wonder if we can somehow avoid the world-writable directory. Maybe per user subdirectories?
Might be also good to mount `RUN_FIREJAIL_DBUS_DIR` noexec.

Reply:
Mounting a tmpfs noexec on `RUN_FIREJAIL_DBUS_DIR` and creating per-uid directories will work fine, I think.

Reply:
However, there is a potential race condition if we mount `RUN_FIREJAIL_DBUS_DIR` in `preproc_build_firejail_dir`: a concurrent firejail process might have created and mounted the directory. `create_empty_dir_as_root` already handles this race by not failing on `EEXIST`, but we might potentially mount tmps twice.

Reply:
Why the tmpfs? I thought we can just remount it as is.

My impression is that we don't need to worry too much about the tiny race window. In the worst case there are one (or few) extra entries in the mount table, there won't be any other real world effect.

Reply:
Ah, indeed, a simple remount works too.

Reply:
But `RUN_FIREJAIL_DBUS_DIR` is not a mountpoint, only `RUN_FIREJAIL_BASEDIR` is, isn't it?

Reply:
> But RUN_FIREJAIL_DBUS_DIR is not a mountpoint

Right, you need to make it a mount point first (bind-mount it onto itself) before you can do the remount.

Reply:
Oh, I see what you mean now.